### PR TITLE
Fix/sync

### PR DESCRIPTION
**Pull request by Imipenem**
**PR Checklist**

 - [x] This comment contains a description of changes (with reason)
 - [ ] Referenced issue is linked
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.rst` is updated

**Description of changes**

* fixed sync command: blacklisted files are now unstaged and changes discarded instead of synced again

* sync PR will now be created from a temporary sync branch

### CODE REVIEW

File: mlf_core/cli.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def sync(project_dir, set_token, pat, username, check_update) -> None:
             sys.exit(1)
         sys.exit(0)
 
-    syncer = TemplateSync(new_template_version='', project_dir=project_dir_path, gh_username=username, token=pat, repo_owner=project_data['github_username'])
+    syncer = TemplateSync(new_template_version='', project_dir=project_dir_path, gh_username=username, token=pat)
```
Comment (Zethson) [inline mlf_core/cli.py:173]: I do not remember this being removed from CT. 
Why can we remove this now?

File: mlf_core/sync/sync.py [hunk 2]
```diff
@@ -61,8 +66,8 @@ def __init__(self, project_dir,
         self.patch_update = patch_update
         self.gh_username = gh_username if gh_username else load_github_username()
         self.token = token if token else decrypt_pat()
-        self.repo_owner = repo_owner if repo_owner else gh_username
         self.dot_mlf_core = {}
+        self.repo_owner = self.gh_username
```
Comment (Zethson) [inline mlf_core/sync/sync.py:70]: Ah, now I see.
Actually I am pretty sure that I ran into some edge case with qube when doing exactly that.

Can you verify that it works for organization repositories?

File: mlf_core/sync/sync.py [hunk 3]
```diff
@@ -97,8 +102,9 @@ def inspect_sync_dir(self):
         if not os.path.exists(os.path.join(str(self.project_dir), '.mlf_core.yml')):
             print(f'[bold red]{self.project_dir} does not appear to contain a .mlf_core.yml file. Did you delete it?')
             sys.exit(1)
-        # store .mlf_core.yml content for later reuse in the dry create run
+            # store .mlf_core.yml content for later reuse in the dry create run
```
Comment (Zethson) [inline mlf_core/sync/sync.py:105]: Why?

File: mlf_core/sync/sync.py [hunk 4]
```diff
@@ -159,14 +168,19 @@ def make_template_project(self):
         Delete all files and make a fresh template.
         """
         print('[bold blue]Creating a new template project.')
-        # dry create run from mlf-core in tmp directory
+        # dry create run from dot_mlf_core in tmp directory
```
Comment (Zethson) [inline mlf_core/sync/sync.py:171]: mlf-core

File: mlf_core/cli.py (hunk 1 above)
Comment (Imipenem) [reply mlf_core/cli.py:173]: I was wondering the same thing, but we removed it in cookietemple as well. Reason being, the repo owner will be directly determined from the .mlf_core.yml , as there we have all info we need and we parse the data anyway.

File: mlf_core/sync/sync.py (hunk 2 above)
Comment (Imipenem) [reply mlf_core/sync/sync.py:70]: I did a test with cookietemple and it worked

File: mlf_core/sync/sync.py (hunk 3 above)
Comment (Imipenem) [reply mlf_core/sync/sync.py:105]: Seems like auto formatting is trolling again

File: mlf_core/sync/sync.py (hunk 2 above)
Comment (Zethson) [reply mlf_core/sync/sync.py:70]: If only I remember which edge case I have in the back of my mind ahhhhhhhhhhh

**Comment by Imipenem** [comment]
May need to update the docs as well regarding sync, I guess @Zethson ?

**Comment by Zethson** [comment]
> May need to update the docs as well regarding sync, I guess @Zethson ?

Care to create an issue? I'll then take care of it.

File: mlf_core/sync/sync.py [hunk 5]
```diff
@@ -294,26 +341,32 @@ def check_sync_level(self) -> bool:
...
             parser = ConfigParser()
             parser.read(f'{self.project_dir}/mlf_core.cfg')
             level_item = list(parser.items('sync_level'))
+            log.debug(f'Parsing level constraint returned: {level_item}.')
             # check for proper configuration if the sync_level section (only one item named ct_sync_level with valid levels major or minor
             if len(level_item) != 1 or 'ct_sync_level' not in level_item[0][0] or not any(level_item[0][1] == valid_lvl for valid_lvl in
                                                                                           ['major', 'minor', 'patch']):
-                print('[bold red]Your sync_level section is missconfigured. Make sure that it only contains one item named mlf_core_sync_level'
-                      ' with only valid levels patch, minor or major!')
+                print('[bold red]Your sync_level section is missconfigured. Make sure that it only contains one item named ct_sync_level with only valid levels'
```
Comment (Zethson) [inline mlf_core/sync/sync.py:353]: `ct_sync_level `

replace with mlf_core_sync_level

File: mlf_core/sync/sync.py [hunk 6]
```diff
@@ -345,30 +401,35 @@ def reset_target_dir(self):
             sys.exit(1)
 
     @staticmethod
-    def update_sync_token(project_name: str, gh_username: str = '') -> None:
+    def update_sync_token(project_name: str, gh_username='') -> None:
```
Comment (Zethson) [inline mlf_core/sync/sync.py:404]: is it not a str?

File: mlf_core/sync/sync.py [hunk 7]
```diff
@@ -159,14 +168,19 @@ def make_template_project(self):
...
-        # dry create run from mlf-core in tmp directory
+        # dry create run from dot_mlf_core in tmp directory
         with tempfile.TemporaryDirectory() as tmpdirname:
-            # COOKIETEMPLE TODO: Refactor this by passing a path param to the create and choose_domain functions, which default to PWD if not passed
             old_cwd = str(Path.cwd())
+            log.debug(f'Saving current working directory {old_cwd}.')
             os.chdir(tmpdirname)
+            log.debug(f'Changed directory to {tmpdirname}.')
+            log.debug(f'Calling choose_domain with {self.dot_mlf_core}.')
+            # choose_domain(path=Path.cwd(), domain=None, dot_cookietemple=self.dot_cookietemple)
```
Comment (Zethson) [inline mlf_core/sync/sync.py:178]: This line must be removed, right?

> choose_domain(domain=None, dot_mlf_core=self.dot_mlf_core)

is below

Comment (Imipenem) [reply mlf_core/sync/sync.py:178]: Yeah, my bad. Was just a reminder that I have to fix this here as well as soon as I add the create output path parameter

File: mlf_core/sync/sync.py (hunk 6 above)
Comment (Imipenem) [reply mlf_core/sync/sync.py:404]: Will fix

**Pull request merged**